### DISCUSSION TITLE
select data

### DISCUSSION THREAD

**Discussion by wibed** [StopNGo/react-proto#4] [General]
on 
```jsx
      const store = initStore(preloadedState)

      /*
        Prefetching with RTK Query:
        - Get data;
        - Put it into the store;
        - Set the store as a preloaded state;
        - Use this store during SSR and for client hydration;
        - Prefetched data will be taken from RTK Query cache in the store.
        Note: Why not just get data during SSR?
        Because rendering will be done before resolving the request Promise.
      */
      await apiRequest(store)
      preloadedState = { ...store.getState() }
```

data is prefetched, as intended by hydration.

now on a more complex case, 

how would one select the data to get prefetched beforehand? 
does the client do any fetching?
how does it work with loadable components?

i am not sure how to apply this to a real case scenario where there are more than one endpoint specified.
what would be your recommendation?
@StopNGo 

ps: 
i ask this in relation to the inability to unsubscribe from the data provided.
```
CAUTION
Manually dispatching an RTKQ request thunk will create a subscription entry, but it's then up to you to [unsubscribe from that data later](https://redux-toolkit.js.org/rtk-query/usage/usage-without-react-hooks#removing-a-subscription) - otherwise the data stays in the cache permanently. In this case, we always need user data, so we can skip unsubscribing.
```

maybe i am not getting a central concept here.

pps:
it could be that `apiRequest.tsx` is just an example on how to incorporate api request on ssr, and is not necessary at all to get ssr going.

**Comment by StopNGo** [comment]
> it could be that apiRequest.tsx is just an example on how to incorporate api request on ssr, and is not necessary at all to get ssr going.

You are absolutely right.
